ic-agent/src/agent/route_provider.rs

```diff
@@ -51,6 +51,15 @@ pub trait RouteProvider: std::fmt::Debug + Send + Sync {
     /// appearing first. The returned vector can contain fewer than `n` URLs if
     /// fewer are available.
     fn n_ordered_routes(&self, n: usize) -> Result<Vec<Url>, AgentError>;
+
+    /// Returns the total number of routes and healthy routes as a tuple.
+    ///
+    /// - First element is the total number of routes available (both healthy and unhealthy)
+    /// - Second element is the number of currently healthy routes, or None if health status information is unavailable
+    ///
+    /// A healthy route is one that is available and ready to receive traffic.
+    /// The specific criteria for what constitutes a "healthy" route is implementation dependent.
+    fn routes_stats(&self) -> (usize, Option<usize>);
```

Comment:
Suggest using a type rather than a tuple
```suggestion
    fn routes_stats(&self) -> RoutesStats;
```

along with something like this
```
pub struct RoutesStats {
    /// total number of routes available (both healthy and unhealthy)
    pub total : usize;

    /// number of currently healthy routes, or None if health status information is unavailable
    ///
    /// A healthy route is one that is available and ready to receive traffic.
    /// The specific criteria for what constitutes a "healthy" route is implementation dependent.
    pub healthy: Option<usize>;
}
```